ariadne/src/Ariadne/UX/CommandHistory.hs

```diff
@@ -27,7 +31,8 @@ openCommandHistory historyFile = do
     counter <- newIORef (-1)
     currentPrefix <- newIORef ""
 
-    unlessM (doesFileExist historyFile) $ writeFile historyFile ""
+    conn <- open historyFile
```

Comment:
[The documentation](https://www.stackage.org/haddock/lts-11.2/sqlite-simple-0.4.14.0/Database-SQLite-Simple.html#v:open) states that every `open` must be closed with a call to `close`, but it's not the case here.
I'd use `withConnection` instead.
The same applies to two other places where `open` is used.